src/test_l1_hdmi_cec_driver.c

```diff
@@ -245,14 +245,19 @@ void test_hdmicec_hal_l1_open_negative( void )
 
     UT_LOG("\n In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
     //Check Null even before calling the positive case
+#ifdef ENABLE_ENHANCED_ERROR_CODE
     result = HdmiCecOpen( NULL );
     if (HDMI_CEC_IO_INVALID_HANDLE  != result) { UT_FAIL("open failed"); }
```

Comment:
if (HDMI_CEC_IO_INVALID_HANDLE  != result) { UT_FAIL("open failed"); }

should be

```c
UT_ASSERT_EQUAL( (HDMI_CEC_IO_INVALID_HANDLE, result );
or
UT_ASSERT( (HDMI_CEC_IO_INVALID_HANDLE == result );
```

Reply:
```c
    result = HdmiCecOpen( &handle );
#ifdef ENABLE_ENHANCED_ERROR_CODE
    if (HDMI_CEC_IO_ALREADY_OPEN != result) { UT_FAIL("open failed"); }
#else
    if (HDMI_CEC_IO_SUCCESS != result) { UT_FAIL("open failed"); }
#endif
```

becomes

```
    result = HdmiCecOpen( &handle );
    if ( feature_enchanged_error_code )
    {
       UT_ASSERT_EQUALS( HDMI_CEC_IO_ALREADY_OPEN, result );
   }
   else
   {
       UT_ASSERT_EQUALS( HDMI_CEC_IO_SUCCESS, result );
   }
```

becomes

```c
 void checkEnchancedError( result, enhanced, old )
{
      if ( feature_enchanged_error_code )
    {
       UT_ASSERT_EQUALS( enhanced, result );
   }
   else
   {
       UT_ASSERT_EQUALS( old, result );
   }
}
```

```c
    result = HdmiCecOpen( &handle );
   checkEnchancedError( result, HDMI_CEC_IO_ALREADY_OPEN, HDMI_CEC_IO_SUCCESS );
```